runtime/ftplugin/query.lua

```diff
@@ -2,5 +2,29 @@
 -- Language:	Tree-sitter query
 -- Last Change:	2022 Mar 29
 
+if vim.b.did_ftplugin == 1 then
+  return
+end
+
+-- Do not set vim.b.did_ftplugin = 1 to allow loading of ftplugin/lisp.vim
+
+-- use treesitter over syntax
+vim.treesitter.start()
+
+-- query linter
+local buf = vim.api.nvim_get_current_buf()
+local query_lint_on = vim.b.lint_on or { 'InsertLeave', 'CursorHoldI', 'BufEnter' }
```

Comment:
If we change some text in insert mode, escape (triggering `InsertLeave`) and there are lint errors, pressing undo won't remove the errors.

I really think we should include `TextChanged` (maybe with a debounce).

Reply:
I think the `check_toplevel` checks could make `TextChanged` a little too noisy. Looking at how we can reduce these. If there's a `check_toplevel` lint, then there's usually something more specific anyway.

Reply:
For now maybe just add `BufWritePre` to the existing list?

Reply:
Let's keep the list short for now; we can always add more events later (which is easier than removing events).

More important is to get the interface right (buffer vs. global variables); my preference is still 1. a global variable for the events (none disables globally)
2. a buffer variable for skipping linting (for the tree, primarily -- we don't even have to expose it for now and treat it as an implementation detail for that purpose only)